src/utils/permissions.ts

```diff
@@ -191,8 +112,134 @@ export async function hasChannelPermissions(
   if (permissions.every((perm) => allowedPermissions.has(perm))) return true;
 
   // Some permission was not explicitly allowed so we default to checking role perms directly
-  const hasPerms = await botHasPermission(guild.id, permissions);
-  return hasPerms;
+  return await calculateServerPermissions(memberID, guild.id, permissions);
+}
+
+export async function calculateServerPermissions(
```

Comment:
This function does not make sense to me. It is called calculateServerPermissions so i would assume this returns to me the perms for this user but in reality it does not.
It returns a boolean a string or a perm. So either this function is named badly or its code is mixed up with something else

Reply:
Suggested rename: checkServerPermissions?